### PR TITLE
have MU_EAT_EMPTY_ARGS

### PR DESCRIPTION
**Pull request by anporumb**

### CODE REVIEW

File: tests/mu_eat_empty_args.c [hunk 1]
```diff
@@ -0,0 +1,200 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
```
Comment (dcristoloveanu) [inline tests/mu_eat_empty_args.c]: >icen [](start = 4, length = 4)

/poetry: mu_empty_args_test.c

File: tests/main.c [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+#include "define_enum_2_test.h"
+#include "mu_eat_empty_args.h"
+
+int main(void)
+{
+    int result;
+    result = run_define_enum_2_test();
+    if (result != 0)
+    {
+        printf("failures in run_define_enum_2_test\n");
```
Comment (dcristoloveanu) [inline tests/main.c]: >printf [](start = 8, length = 6)

(void)

File: tests/main.c [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        return result;
+    }
+
+    result = run_mu_eat_empty_args_test();
+    if (result != 0)
+    {
+        printf("failures in run_mu_eat_empty_args_test\n");
```
Comment (dcristoloveanu) [inline tests/main.c]: >printf [](start = 8, length = 6)

(void)

File: inc/azure_macro_utils/macro_utils.h [hunk 4]
```diff
@@ -12696,6 +12696,1728 @@ MU_IF(X, "true", "false") => "true"
 #define MU_IF1(trueBranch, falseBranch) trueBranch
 
 
+/*the following macro want to eat empty arguments from a list */
+/*examples:                                                   */
+/*MU_EAT_EMPTY_ARGS(, , X) expands to X                       */
+
+#define MU_EAT_EMPTY_ARGS0() 
+#define MU_EAT_EMPTY_ARGS1(x) x
```
Comment (dcristoloveanu) [inline inc/azure_macro_utils/macro_utils.h]: Seems a little bit much to generate each one. I would highly consider using the existing infra for looping (FOR_EACH).

I tested this and it passed the tests:

#define MU_EXPAND_TO_NOTHING(arg)
#define MU_EAT_EMPTY_ARG(arg_count, arg) MU_IF(MU_ISEMPTY(arg),,arg) MU_IF(MU_ISEMPTY(arg),MU_EXPAND_TO_NOTHING,MU_IFCOMMALOGIC)(MU_DEC(arg_count))
#define MU_EAT_EMPTY_ARGS(...) MU_FOR_EACH_1_COUNTED(MU_EAT_EMPTY_ARG, __VA_ARGS__)

File: tests/mu_eat_empty_args.c [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+{                                                                                                       \
+    if (!(condition))                                                                                   \
+    {                                                                                                   \
+        printf("condition \"" #condition "\" in LINE=" MU_TOSTRING(__LINE__) " was UNEXPECTEDLY FALSE\n");          \
+        return MU_FAILURE;                                                                              \
+    }                                                                                                   \
+} while (0)
+
+
+int run_mu_eat_empty_args_test(void)
```
Comment (dcristoloveanu) [inline tests/mu_eat_empty_args.c]: >run_mu_eat_empty_args_test [](start = 4, length = 26)

Cheers. This was very useful to just test my alternative implementation.

File: tests/mu_eat_empty_args.c (hunk 1 above)
Comment (anporumb) [reply tests/mu_eat_empty_args.c]: renamed

---
In reply to: [270706181](https://github.com/Azure/azure-macro-utils-c/pull/11#discussion_r270706181) [](ancestors = 270706181)

File: tests/main.c (hunk 2 above)
Comment (anporumb) [reply tests/main.c]: voided

---
In reply to: [270706199](https://github.com/Azure/azure-macro-utils-c/pull/11#discussion_r270706199) [](ancestors = 270706199)

File: tests/main.c (hunk 3 above)
Comment (anporumb) [reply tests/main.c]: above and everyhere.

---
In reply to: [270706215](https://github.com/Azure/azure-macro-utils-c/pull/11#discussion_r270706215) [](ancestors = 270706215)

File: inc/azure_macro_utils/macro_utils.h (hunk 4 above)
Comment (anporumb) [reply inc/azure_macro_utils/macro_utils.h]: I bow before the master.

---
In reply to: [270708972](https://github.com/Azure/azure-macro-utils-c/pull/11#discussion_r270708972) [](ancestors = 270708972)

File: tests/mu_eat_empty_args.c (hunk 5 above)
Comment (anporumb) [reply tests/mu_eat_empty_args.c]: Cheers!

---
In reply to: [270709079](https://github.com/Azure/azure-macro-utils-c/pull/11#discussion_r270709079) [](ancestors = 270709079)

**Pull request merged**